src/ParallelAlgorithms/SurfaceFinder/Python/FindRadialSurface.py

```diff
@@ -0,0 +1,274 @@
+    surface_radii.fill(np.nan)
+    filled = np.zeros(len(block_logical_coords), dtype=bool)
+    for element, data in iter_elements(
+        open_volfiles(h5_files, subfile_name, obs_id),
+        obs_ids=obs_id,
+        tensor_components=[var_name],
+    ):
+        # Find the radial rays that go through this element
+        angular_coords = []
+        offsets = []
+        for i, block_logical_coord in enumerate(block_logical_coords):
+            if filled[i]:
+                continue
+            block_id = block_logical_coord.id.get_index()
+            # Radial neighbors that are wedges are also valid because they keep
+            # the angular logical coordinates the same
+            block_neighbors = [
+                block_neighbor_id
+                for direction, block_neighbor_id in domain.blocks[
+                    block_id
+                ].neighbors.items()
+                if direction.dimension == 2
+                and domain.blocks[block_neighbor_id].has_distorted_frame()
+            ]
+            valid_blocks = [block_id, *block_neighbors]
+            if element.id.block_id in valid_blocks:
+                continue
+            angular_block_logical = np.array(block_logical_coord.data)[:2]
+            angular_element_id = ElementId[2](
+                element.id.block_id, element.id.segment_ids[:2]
+            )
+            angular_element_logical = element_logical_coordinates(
+                angular_block_logical, angular_element_id
+            )
+            if angular_element_logical is None:
+                continue
+            angular_coords.append(angular_element_logical)
+            offsets.append(i)
+        if len(angular_coords) == 0:
+            continue
+        # Find the surface in this element along each ray
+        angular_coords = np.array(angular_coords).T
+        surface_logical_radii = spectre.SurfaceFinder.find_radial_surface(
+            data[0], target, element.mesh, angular_coords
+        )
+        # For those rays where we found the surface in this element, map the
+        # surface radius to inertial coordinates
+        for offset, surface_logical_radius, angular_element_logical in zip(
+            offsets, surface_logical_radii, angular_coords.T
+        ):
+            if surface_logical_radius is None:
+                continue
+            surface_logical_point = [
+                *angular_element_logical,
+                surface_logical_radius,
+            ]
+            surface_radii[offset] = np.linalg.norm(
+                np.array(
+                    element.map(
+                        np.array(surface_logical_point).reshape((3, 1)),
+                        time,
+                        functions_of_time,
+                    )
+                )[:, 0]
+            )
+            filled[offset] = True
+        if np.all(filled):
+            break
+    if not np.all(filled):
+        raise ValueError("Unable to find the surface.")
```

Comment:
Would there be any value in printing out any additional info? Or maybe a verbosity flag to allow printing more? E.g. did no rays succeed or only some not succeed?

Reply:
Changed to this:

```
ValueError: Unable to find the surface where Field = 0.2 with center [0.0, 0.0, 0.0]. Missing values for 325 / 325 
radial rays. One of the missing rays goes through: [0.01771545 0.         0.09841831]
```